x-pack/plugins/cloud_security_posture/server/routes/configuration/update_rules_configuration.ts

```diff
@@ -77,57 +69,197 @@ const getEnabledRulesByBenchmark = (rules: SavedObjectsFindResponse<CspRule>['sa
+      packagePolicyId,
+      policyId,
+    }),
+    fields: RUNTIME_CFG_FIELDS,
+    searchFields: ['name'],
+    perPage: 10000,
+  });
+
+/**
+ * gets rules by id of a package policy with fields required for updating SO object
+ */
+const getPartialCspRulesById = (
+  soClient: CspApiRequestHandlerContext['soClient'],
+  ruleIds: string[]
+) =>
+  soClient.bulkGet<SavedObjectRuleUpdatePayload>(
+    ruleIds.map((ruleId) => ({
+      id: ruleId,
+      type: CSP_RULE_SAVED_OBJECT_TYPE,
+      fields: ['enabled'],
+    }))
+  );
+
+const getRuntimeCfgVarValue = (rules: Array<SavedObject<PackagePolicyRuleUpdatePayload>>) =>
+  yaml.safeDump(createRulesConfig(rules));
+
+/**
+ * Updates the package policy vars object with a new value for the runtimeCfg key
+ * @internal
+ * */
+export const updatePackagePolicyVars = async ({
+  packagePolicyService,
+  packagePolicy,
+  esClient,
+  soClient,
+  user,
+}: {
+  packagePolicyService: PackagePolicyServiceInterface;
+  packagePolicy: PackagePolicy;
+  esClient: ElasticsearchClient;
+  soClient: SavedObjectsClientContract;
+  user: AuthenticatedUser | null;
+}): Promise<PackagePolicy> => {
+  const packagePolicyRulesSO = (
+    await getAllPackagePolicyCspRules(soClient, packagePolicy.id, packagePolicy.policy_id)
+  ).saved_objects;
+
   return packagePolicyService.update(
     soClient,
     esClient,
     packagePolicy.id,
-    updatedPackagePolicy,
-    options
+    getUpdatedPackagePolicy(packagePolicy, getRuntimeCfgVarValue(packagePolicyRulesSO)),
+    user ? { user } : undefined
   );
 };
 
+/**
+ * Keeps only fields we want/need to use for updating a rule saved-object
+ */
+const getRuleUpdateFields = ({
+  id,
+  type,
+  attributes,
+  references,
+}: SavedObject<SavedObjectRuleUpdatePayload>) => ({ id, type, attributes, references });
+
+/**
+ * Gets current rules with next values for 'enabled' property
+ */
+const getNextCspRules = (
+  current: Array<SavedObject<SavedObjectRuleUpdatePayload>>,
+  rulesToChange: UpdateRulesConfigBodySchema['rules']
+) => {
+  const currentRulesMap = Object.fromEntries(
+    current.map((rule) => [rule.id, getRuleUpdateFields(rule)])
+  );
+  return rulesToChange.map((rule) => ({
+    ...currentRulesMap[rule.id],
+    attributes: {
+      ...currentRulesMap[rule.id].attributes,
+      enabled: rule.enabled,
+    },
+  }));
+};
+
+/**
+ * Combines updating rules in saved-objects and package policy vars into a single operation
+ * @internal
+ */
+export const updateRulesById = async (
+  { soClient, packagePolicyService, esClient, user, logger }: CspApiRequestHandlerContext,
+  { rules: rulesToChange, package_policy_id: packagePolicyId }: UpdateRulesConfigBodySchema
+): Promise<PackagePolicy> => {
+  logger.info(`Start updating rules: ${JSON.stringify(rulesToChange)}`);
+
+  const currentRules = (
+    await getPartialCspRulesById(
+      soClient,
+      rulesToChange.map((rule) => rule.id)
+    )
+  ).saved_objects;
```

Comment:
What if the saved object update fail? it still continue to update the package policy?